server/src-lib/Hasura/GraphQL/Resolve/Introspect.hs

```diff
@@ -230,22 +270,25 @@ namedTypeR nt fld = do
 
 namedTypeR'
   :: ( MonadReader r m, Has TypeMap r
-     , MonadError QErr m)
+     , MonadError QErr m, MonadReusability m)
   => Field
   -> TypeInfo
   -> m J.Object
-namedTypeR' fld = \case
-  TIScalar colTy        -> scalarR colTy fld
-  TIObj objTyInfo       -> objectTypeR objTyInfo fld
-  TIEnum enumTypeInfo   -> enumTypeR enumTypeInfo fld
-  TIInpObj inpObjTyInfo -> inputObjR inpObjTyInfo fld
-  TIIFace iFaceTyInfo   -> ifaceR' iFaceTyInfo fld
-  TIUnion unionTyInfo   -> unionR unionTyInfo fld
+namedTypeR' fld tyInfo = do
+  -- Now fetch the required type information from the corresponding
```

Comment:
This comment [may not be needed](https://github.com/hasura/graphql-engine/blob/master/server/STYLE.md#inline-comments) but I don't feel strongly about it, so your call.

Reply:
This comment is here to help any future readers of the code who may be confused, as I was initially, about what is going on here. I like to document things that are obvious in hindsight :-)